### ISSUE TITLE
shadowed label false positive in closure

### ISSUE THREAD

**Issue by durka** [rust-lang/rust#31754]
It seems that #27093 did not fully fix #25343. [The following code](http://is.gd/qA7fIE) reports a spurious warning on `'b`:

```rust
'a: loop { break; }
'b: loop { break; }

// no warning
fn does_not_shadow() {
    'a: loop {}
}

// false positive
let also_does_not_shadow = || {
    'b: loop {}
};
```

Stable, beta, nightly.
